wolfcrypt/src/ecc.c

```diff
@@ -7545,8 +7548,24 @@ int wc_ecc_gen_deterministic_k(const byte* hash, word32 hashSz,
         return BAD_FUNC_ARG;
     }
 
-    if (hashSz != WC_SHA256_DIGEST_SIZE) {
-        WOLFSSL_MSG("Currently only SHA256 digest is supported");
+    /* if none is provided then detect has type based on hash size */
+    if (hashType == WC_HASH_TYPE_NONE) {
+        if (hashSz >= 64) {
+            hashType = WC_HASH_TYPE_SHA512;
+        }
+        else if (hashSz >= 48) {
+            hashType = WC_HASH_TYPE_SHA384;
+        }
+        else {
+            hashType = WC_HASH_TYPE_SHA256;
+        }
+    }
+
+    /* For deterministic k only SHA2-256, SHA2-384 and SHA2-512 are supported */
```

Comment:
@SparkiDev do we need this restriction? Seems like any supported HMAC algo would be fine. As long as we properly fail later if the hash type is unsupported or invalid. Thoughts?

Reply:
Yep. We can remove this now.